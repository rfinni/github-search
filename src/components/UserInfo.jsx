import React from 'react';
import PropTypes from 'prop-types';
import UserImg from './UserImg';
import UserData from './UserData';

const UserInfo = ({
  user,
}) => (
  <div className="user-info">
    <a rel="noopener noreferrer" href={user.url} target="_blank">
      <UserImg user={user} />
      <UserData user={user} />
    </a>
  </div>
);

UserInfo.propTypes = {
  user: PropTypes.object.isRequired,
};

export default UserInfo;
