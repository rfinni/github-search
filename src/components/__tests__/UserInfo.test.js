import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from '../UserInfo';
import { stubUser } from '../../../testing/stubs';

describe('UserInfo.jsx', () => {
  const wrapper = renderer.create(<UserInfo user={stubUser} />).toJSON();
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

