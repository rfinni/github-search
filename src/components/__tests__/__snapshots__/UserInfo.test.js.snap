// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo.jsx should render correctly 1`] = `
<div
  className="user-info"
>
  <a
    href="https://api.github.com/users/rfinni"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="Ryan Finni"
        className="user-image"
      />
    </div>
    <div
      className="user-data"
    >
      <span
        className="name"
      >
        Ryan Finni
         (
        rfinni
        )
      </span>
      <span>
        Chicago, IL
      </span>
      <span>
        Joined GitHub in 
        NaN
      </span>
    </div>
  </a>
</div>
`;
